.bazelrc

```diff
@@ -9,31 +9,43 @@ build:wasm32 --cpu=wasm32
 # Use a different symlink prefix for wasm-based artifacts.
 build:wasm32 --symlink_prefix=bazel-wasm-
 
-build:clang --crosstool_top=//toolchain:clang
-build:clang --host_crosstool_top=@bazel_tools//tools/cpp:toolchain
-build:clang --cpu=k8
+# Common options for clang-based compilation
 # Force the usage of stack protectors for all functions.
-build:clang --cxxopt="-fstack-protector-all"
+build:clang-base --cxxopt="-fstack-protector-all"
 # Pretty colours.
-build:clang --cxxopt="-fcolor-diagnostics"
+build:clang-base --cxxopt="-fcolor-diagnostics"
 # Don't omit the frame pointer, useful for debug.
-build:clang --cxxopt="-fno-omit-frame-pointer"
+build:clang-base --cxxopt="-fno-omit-frame-pointer"
 # Enable all warnings.
-build:clang --cxxopt="-Wall"
+build:clang-base --cxxopt="-Wall"
 # Enable extra warnings.
-build:clang --cxxopt="-Wextra"
+build:clang-base --cxxopt="-Wextra"
 # Simple but useful static checking to detect potential race conditions.
-build:clang --cxxopt="-Wthread-safety"
+build:clang-base --cxxopt="-Wthread-safety"
 # Treat warnings as errors.
-build:clang --cxxopt="-Werror"
+build:clang-base --cxxopt="-Werror"
 # Disable some warnings:
 # protobuf generates unused parameters, so skip this for now.
-build:clang --cxxopt="-Wno-unused-parameter"
+build:clang-base --cxxopt="-Wno-unused-parameter"
 # https://stackoverflow.com/questions/1538943
-build:clang --cxxopt="-Wno-missing-field-initializers"
-build:clang --cxxopt="-Wno-error=sign-compare"
+build:clang-base --cxxopt="-Wno-missing-field-initializers"
+build:clang-base --cxxopt="-Wno-error=sign-compare"
+# Allow dependencies to make use of library functions marked deprecated,
+# particularly on OS X.
+build:clang-base --cxxopt="-Wno-error=deprecated-declarations"
+# Workaround for missing thread-local reference in gRPC.
+# https://github.com/bazelbuild/bazel/issues/4341
+build:clang-base --copt=-DGRPC_BAZEL_BUILD
 # Use a different symlink prefix for clang-based artifacts.
-build:clang --symlink_prefix=bazel-clang-
+build:clang-base --symlink_prefix=bazel-clang-
+
+build:clang --config=clang-base
+build:clang --crosstool_top=//toolchain:clang
+build:clang --host_crosstool_top=@bazel_tools//tools/cpp:toolchain
+build:clang --cpu=k8
+
+# Darwin's default compiler is clang, so no need for a crosstool
+build:darwin --config=clang-base
```

Comment:
I think the advantage of specifying a "cross" compiler was that it will force a certain version of clang (the one we download) instead of the one available on the system. Not really a big problem, depends how much we want to control that.

Reply:
Having a known version would be nice, but the wrapper scripts in `toolchain/` didn't seem to quite work for a MacOS build, and just falling back to the native toolchain was the easy way out.

I can revisit what was going wrong with `--config=clang` but I might need your help with toolchain wrangling&hellip;